tests/migration/conftest.py

```diff
@@ -0,0 +1,44 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Any, Dict, Generator, Union
+
+import pytest
+import pytest_alembic
+from pytest_alembic.config import Config
+
+if TYPE_CHECKING:
+    from pytest_alembic import MigrationContext
+    from sqlalchemy.engine import Engine
+
+    import alembic.config
+    from tests.fixtures.database import ApplicationFixture, DatabaseFixture
+
+pytest_plugins = [
+    "tests.fixtures.api_config",
+    "tests.fixtures.database",
+]
+
+
+@pytest.fixture
+def alembic_engine(
+    application: ApplicationFixture, database: DatabaseFixture
+) -> Engine:
+    """
+    Override this fixture to provide pytest-alembic powered tests with a database handle.
+    """
+    return database._engine
+
+
+@pytest.fixture
+def alembic_runner(
+    alembic_config: Union[Dict[str, Any], alembic.config.Config, Config],
+    alembic_engine: Engine,
+) -> Generator[MigrationContext, None, None]:
+    """
+    Override this fixture to make sure that we stamp head. Since this is how out database
+    is initialized. The normal fixtures assume you start from an empty database.
+    """
+    config = Config.from_raw_config(alembic_config)
+    with pytest_alembic.runner(config=config, engine=alembic_engine) as runner:
+        runner.command_executor.stamp("head")
```

Comment:
While using the alembic_runner fixture, according to the docs one of the useful methods for custom tests is the `alembic_runner.migrate_up_to` method, which will not work as intended if we've already stamped the migration state at `head`. I think.
https://pytest-alembic.readthedocs.io/en/latest/api.html#alembic-runner

Reply:
@RishiDiwanTT agreed that `migrate_up_to` won't work, but there is a similar function `migrate_down_to` and I think that will work for our testing. 

I wrote a test using pytest alembic for #1063 here: 
https://github.com/ThePalaceProject/circulation/blob/8f339a671cf94b36d53d8fc8359cab1efc1e820d/tests/migration/test_20230510_a9ed3f76d649.py#L59

And it seemed to work okay using `migrate_down_to`,`migrate_down_one` and `migrate_up_one` to do some testing.

Reply:
Perfect :+1: